src/System.Private.CoreLib/src/System/Reflection/RuntimeFieldInfo.cs

```diff
@@ -68,6 +68,7 @@ public override Type DeclaringType
         public sealed override bool HasSameMetadataDefinitionAs(MemberInfo other) => HasSameMetadataDefinitionAsCore<RuntimeFieldInfo>(other);
 
         public override Module Module { get { return GetRuntimeModule(); } }
+        public override bool IsCollectible => DeclaringType?.IsCollectible ?? false;
```

Comment:
In MemberInfo.cs you default non-standard MemberInfo's to returning true, but here you have them return false. What's the rationale?

Reply:
`DeclaringType` is `null` in the event that `m_reflectedTypeCache.IsGlobal` is `true`.  I was under the impression that this means the type belonged to the global loader allocator and therefore wasn't collectible.  The default value being true was based on a comment from Jan in a previous review.

Reply:
`DeclaringType is null` means that the method is global. You cannot write such methods in C# today.

The correct thing to do for  `DeclaringType is null` would be to fetch the `IsCollectible` from get the containing module. It may be easier to just call into the runtime like you are doing it for RuntimeMethodInfo.

Reply:
Would it be sufficient to simply check `m_declaringType`'s collectilblity, like so:
```suggestion
        public override bool IsCollectible => m_declaringType.IsCollectible;
```

Reply:
Yes, this looks good.